rskj-core/src/main/java/org/ethereum/vm/VM.java

```diff
@@ -1553,25 +1549,38 @@ protected void doCALL(){
 
         program.memoryExpand(outDataOffs, outDataSize);
 
-        MessageCall msg = new MessageCall(
+        return new MessageCall(
                 MsgType.fromOpcode(op),
                 DataWord.valueOf(calleeGas), codeAddress, value, inDataOffs, inDataSize,
                 outDataOffs, outDataSize);
+    }
 
-        callToAddress(codeAddress, msg);
-
-        program.step();
+    private DataWord calculateCallValue(ActivationConfig.ForBlock activations) {
```

Comment:
I would call this something different from calculate. calculate makes me think this is a transparently referential function, but because this can do a program.stackPop(), it's actually imperative. In fact, it almost always does a stackPop, it seems, except in two exceptional op codes. So it makes sense to me to call it something like popCallValue(), or even maybePopCallValue().